server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -1789,8 +1795,23 @@ public TemplatePermissionsResponse createTemplatePermissionsResponse(ResponseVie
         // Set accounts
         List<String> projectIds = new ArrayList<String>();
         List<String> regularAccounts = new ArrayList<String>();
+        final boolean isRootAdmin = _accountMgr.isRootAdmin(CallContext.current().getCallingAccountId());
         for (String accountName : accountNames) {
-            Account account = ApiDBUtils.findAccountByNameDomain(accountName, templateOwner.getDomainId());
+            // Root Admins should be allowed to access permissions for all accounts across domains
+            Account account = null;
+            if (isRootAdmin) {
+                List<AccountVO> accounts = accountDao.findAccountsLike(accountName);
+                if (accounts != null && accounts.size() > 0) {
+                    account = accounts.get(0);
```

Comment:
@anuragaw could this cause regression or issues, this is always picking up the first account in the list; should it be something like - if user is root admin, then all enable/permit for all accounts matching the name using something like `accountIds.add(permittedAccount.getId());` in the if block; and refactor suitably?

Reply:
Should this find account using name and domain?

Reply:
No longer part of PR as above.